### ISSUE TITLE
keep_order doesn't work as expected

### ISSUE THREAD

**Issue by rmzelle** [daler/gffutils#46]
See https://gist.github.com/rmzelle/5697c87edf376f17fd63

There are two GFF files. The stringified output of all features seems to put the first attribute of the first feature in front, even though I have `keep_order` set to "True". Tested with gffutils 0.8.3 and Python 2.7.9.

**Comment by rmzelle** [comment]
On a second thought, maybe this is the expected behavior? Is the attribute order always determined for the entire GFF db as a whole, instead of on a per-feature level? Would setting `keep_order` to "True" on all features individually change the behavior at all?

In my examples above, the original attribute order is not consistent between features, e.g.:

```
ID=38858;Name=38858
Name=MPH3;...;...;ID=30146
```

It's not a big deal, since I can just run my GFF file through gffutils to sort the attributes first, before making my genome annotation changes, but maybe the documentation can be clarified a little.

**Comment by daler** [comment]
Yep, this is a known issue and basically is a result of a compromise I made during performance optimization.

Discussed a little bit at https://github.com/daler/gffutils/issues/20#issuecomment-17973528 and summarized at http://pythonhosted.org/gffutils/developers.html#little-things. Quoting the latter.

> * I really wanted to maintain round-trip invariance: importing into a db and then getting the features back out should not change them at all. That’s where the dialect comes into play – it specifies the format of the attributes string, which is the trickiest thing to get right.
* at first, I was keeping track of the order of attributes in an OrderedDict. Benchmarking with 1M+ line files showed that this was slow. So now the attributes are stored as plain ol’ dicts, and information about the order of attributes is stored only once: in the db’s dialect. While features with different orders of attributes (on one line “gene_id” comes first; on another line “Name” comes first) will be round-trip invariant, this should at least hold for most cases. I figured it was a good compromise.

I added more information to the `gffutils.create_db` docstring to help clarify this.
